resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/AliasManager.java

```diff
@@ -134,7 +133,7 @@ class LocationMap {
 		/**
 		 * Map of FileStore-&gt;IResource OR FileStore-&gt;ArrayList of (IResource)
 		 */
-		private final SortedMap<IFileStore, Object> map = new TreeMap<>(IFileStore::compareTo);
+		private final Map<IFileStore, Object> map = new HashMap<>(10_000);
```

Comment:
Form the javadoc

>  Iteration over collection views requires time proportional to the "capacity" of the HashMap instance (the number of buckets) plus its size (the number of key-value mappings). Thus, **it's very important not to set the initial capacity too high** (or the load factor too low) **if iteration performance is important**. 

Maybe we should start smaller (or use a system property to control the inital size)